pkg/local_object_storage/writecache/delete.go

```diff
@@ -11,41 +14,71 @@ import (
 //
 // Returns an error of type apistatus.ObjectNotFound if object is missing in write-cache.
 func (c *cache) Delete(addr oid.Address) error {
-	c.modeMtx.RLock()
-	defer c.modeMtx.RUnlock()
-	if c.readOnly() {
-		return ErrReadOnly
-	}
-
+	// Pre-check object location. If object is missing, we return immediately
+	// regardless of current cache mode. Otherwise, we need to check writing
+	// mode for removal.
+	var exists, inDB bool
```

Comment:
Can we just check for `ErrReadOnly` on the caller side?

Reply:
So we fail with read-only error even if object is missing?

Reply:
Yes, I think it is ok. Well, we can also perform `Exists` on the caller side or filter `ErrReadOnly`.

Reply:
So do you suggest to:
1. exclude `ErrReadOnly` from shard logs
2. pre-check object existence in shard layer as you said in the issue
> I suggest lowering the error level here and also check if an object exists first.

Reply:
Yes, `ErrReadOnly` can be excluded.
Also IMO it can be returned befor existence check.